util/paywall.go

```diff
@@ -592,3 +613,56 @@ func FetchTxsForAddressNotBefore(address string, notBefore int64) ([]TxDetails,
 
 	return targetTxs, nil
 }
+
+// FetchTx fetches a given transaction based on the provided txid.
+func FetchTx(address, txid string) ([]TxDetails, error) {
```

Comment:
This is still using the insight api.

Reply:
Yah there is an outstanding PR that I was going to let handle that removal to avoid overlap.

Reply:
That works.

Reply:
That PR has been merged now so you can update accordingly.